hw3_2/hw3-2/Program.cs

```diff
@@ -0,0 +1,452 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace hw3_2
+{
+    class Program
+    {
+        static async Task Main(string[] args)
+        {
+            string path;
+            string argument;
+            if (args.Length != 2)
+            {
+                Console.WriteLine(
+                    "Few arguments! Enter file path and \"-c\" if you want to compress text\nor \"-u\" if decompress");
+                return;
+            }
+
+            path = args[0];
+            argument = args[1];
+            if (argument != "-c" && argument != "-u")
+            {
+                Console.WriteLine(
+                    "Incorrect argument. Enter file path and \"-c\" if you want to compress text\nor \"-u\" if decompress");
+                return;
+            }
+
+            if (!File.Exists(path))
+            {
+                Console.WriteLine("Incorrect file path!");
+                return;
+            }
+
+            Lzw lzw = new();
+            if (argument == "-c")
+            {
+                string? text = "";
+
+                using (StreamReader reader = new StreamReader(path))
+                {
+                    if ((text = await reader.ReadToEndAsync()) == null)
+                    {
+                        Console.WriteLine("Empty file!");
+                        return;
+                    }
+                }
+
+                bool isEmpty;
+                List<Byte[]> listOfCodes = lzw.LzwCompression(text, out isEmpty);
+                if (isEmpty)
+                {
+                    Console.WriteLine("Empty file!");
+                    return;
+                }
+
+                string newName = path + ".zipped";
+                using (FileStream zippedFile = File.Create(newName))
+                {
+                    if (!File.Exists(newName))
+                    {
+                        Console.WriteLine("Incorrect file path!");
+                        return;
+                    }
+
+                    foreach (byte[] code in listOfCodes)
+                    {
+                        zippedFile.Write(code, 0, code.Length);
+                    }
+                }
+
+                Console.WriteLine($"Compression ratio: {4 * listOfCodes.Count / (float)(2 * text.Length)}");

```

Comment:
```suggestion
                Console.WriteLine($"Compression ratio: {listOfCodes.Count / text.Length}");
```
Не понял назначение этих констант. Кажется, что достаточно размер результата компрессии поделить на размер исходного файла. Если я не прав, напишите тут.
И да, если используете магические константы, смысл которых неочевиден - выносите их в именованные константы.

Reply:
А, это количества бит для инта и для символа.
Точно надо в константы.
А ещё лучше, вынести вычисление коэффициента в отдельный метод.